cucumber/authenticators_azure/features/support/authn_azure_helper.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Utility methods for Azure authenticator
+
+module AuthnAzureHelper
+  include AuthenticatorHelpers
+
+  def set_azure_provider_uri_variable(value = azure_provider_uri)
+    set_azure_variable("provider-uri", value)
+  end
+
+  def retrieve_azure_access_token
+    command = "curl 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F' -H Metadata:true -s | jq -r '.access_token'"
+    @azure_token = run_command_in_machine(azure_machine_ip, azure_machine_username, azure_machine_password, command)
+  rescue => err
+    raise "Failed to fetch azure token with reason: #{err}"
+  end
+
+  def authenticate_azure_token(service_id:, account:, username:, azure_token: @azure_token)
+    username = username.gsub("/", "%2F")
+    path = "#{conjur_hostname}/authn-azure/#{service_id}/#{account}/#{username}/authenticate"
+
+    payload = {}
+    payload["jwt"] = azure_token
+
+    post(path, payload)
+  end
+
+  private
+
+  def set_azure_variable(variable_name, value)
```

Comment:
What do you think about renaming this to `create_and_set_azure_variable`.  Using `set` makes me think of setters, which this method is not 😄.